tests/topotests/ospf_te_topo1/test_ospf_te_topo1.py

```diff
@@ -70,30 +70,24 @@ def build_topo(tgen):
     "Build function"
 
     # Create 4 routers
-    for routern in range(1, 5):
-        tgen.add_router("r{}".format(routern))
+    r1 = tgen.add_router("r1")
+    r2 = tgen.add_router("r2")
+    r3 = tgen.add_router("r3")
+    r4 = tgen.add_router("r4")
 
     # Interconect router 1 and 2 with 2 links
-    switch = tgen.add_switch("s1")
-    switch.add_link(tgen.gears["r1"])
-    switch.add_link(tgen.gears["r2"])
-    switch = tgen.add_switch("s2")
-    switch.add_link(tgen.gears["r1"])
-    switch.add_link(tgen.gears["r2"])
+    tgen.add_link(r1, r2, ifname1="eth0", ifname2="eth0")
```

Comment:
Can you continue to use the `r1-eth0`, `r2-eth0` naming scheme rather than removing the `rX-` prefix? If you do that here and below, then none of the config files need to change and this PR gets a lot smaller -- from 10 files changed to 2.

Reply:
Also what are the default names assigned if you don't specify the `ifname[12]` keywords?